2-Operating-Systems/1-arch.md

```diff
@@ -0,0 +1,459 @@
+   4. Ensure USB drives will boot by default
+   5. Reboot into Arch Linux installation environment
+
+4. If needed: Set desired keyboard layout (if not US standard)
+   
+   Example: German
+   
+   ```
+   loadkeys de-latin1
+   ```
+
+5. If needed: Change rotation   
+   
+   ```
+   echo 1 > /sys/class/graphics/fbcon/rotate_all
+   ```
+   
+   Note: May need to use 2 or 3 depending on starting state
+
+6. If needed: Increase font size
+   
+   ```
+   setfont /usr/share/kbd/consolefonts/latarcyrheb-sun32.psfu.gz
+   ```
+   
+   Note: This is the current largest font
+
+7. Connect to the internet
+   
+   1. Select and join a wireless network with `wifi-menu`
+   2. Verify connection with `ping 1.1.1.1`
+
+8. Update system clock
+   
+   ```
+   timedatectl set-ntp true
+   ```
+
+9. Set up encrypted disk
+   
+   1. Determine target root device with `lsblk`. Example: /dev/mmcblk0
+   
+   2. Create all partitions
+      
+      TL:DR;
+      
+      ```
+      sgdisk -Zo -n 1:2048:+512M -t 1:EF00 -c 1:boot -N 2 -t 2:8300 -c 2:root /dev/mmcblk0
+      ```
+      
+      Explanation:
+      
+      ```
+      sgdisk \
+       -Zo             `# zero out any existing partitions` \
+       -n 1:2048:+512M `# create new 512M partition for boot` \
+       -t 1:EF00       `# set type to EFI` \
+       -c 1:boot       `# set comment/label to "boot"` \
+       -N 2            `# create second partition filling disk` \
+       -t 2:8300       `# set type Linux` \
+       -c 2:root       `# set comment/label to "root"` \
+       /dev/mmcblk0     # writing to your target drive
+      ```
+      
+      Note: As an alternative, you can graphically partition using `cgdisk`
+   
+   3. Format the EFI boot partition as FAT32
+      
+      ```
+      mkfs.vfat -F32 /dev/mmcblk0p1
+      ```
+   
+   4. Encrypt and format the root partition
+      
+      ```
+      # encrypt root partition with passphrase
+      cryptsetup -y -v luksFormat --type luks2 /dev/mmcblk0p2
+      
+      # decrypt drive and expose as /dev/mapper/cryptroot
+      cryptsetup open /dev/mmcblk0p2 cryptroot
+      
+      # make journaled ext4 partition in decrypted root device
+      mkfs.ext4 -j /dev/mapper/cryptroot
+      ```
+
+10. Mount the file systems
+    
+    ```
+    mount /dev/mapper/cryptroot /mnt
+    mkdir /mnt/boot
+    mount /dev/mmcblk0p1 /mnt/boot
+    ```
+
+11. Select the Mirrors
+    
+    ```
+    pacman -Sy reflector
+    reflector --age 12 --protocol https --sort rate --save /etc/pacman.d/mirrorlist
+    ```
+    
+    Note: This avoids manual mirror sorting.
+
+12. Install packages
+    
+    1. Install packages
+    
+    ```
+    pacstrap /mnt base sudo dialog wpa_supplicant iw vim git \ 
+      pcsclite libu2f-host chromium arandr compton i3-wm i3status \   dmenu kitty nitrogen slock xorg xorg-xinit xf86-video-intel \   ccid opensc openssh haveged pulseaudio pulseaudio-alsa \    pulsemixer
+    ```
+    
+    Explanation:
+    
+    ```
+    pacman -S \
```

Comment:
the command above for installing packages is `pacstrap` but the explanation uses `pacman`